### PR TITLE
Build MCNP Shared Library

### PR DESCRIPTION
**Pull request by ejwilson3**
This pull request is focused on the functionality of the code to turn MCNP2CAD into a library able to be called by a Trelis-Plugin.  There is another pull request planned for the future with fixes to formatting and commenting issues.

### CODE REVIEW

File: MCNPInput.cpp [hunk 1]
```diff
@@ -54,6 +54,7 @@ static int makeint( const std::string& token ){
   char* end;
   int ret = strtol(str, &end, 10);
   if( end != str+token.length() ){
+    record << "Warning: string [" << token << "] did not convert to int as expected." << std::endl;
     std::cerr << "Warning: string [" << token << "] did not convert to int as expected." << std::endl;
```
Comment (sjackso) [inline MCNPInput.cpp]: :neckbeard: The ghost of hackers past moans:

These duplicated printouts make me sad.  Consider defining
```c
// Note: this macro will evaluate its argument expression twice!
#define NOISY_WARNING(OUTPUT_EXPR) do{ \
   std::cerr << "Warning: " << OUTPUT_EXPR << std::endl; \
   record    << "Warning: " << OUTPUT_EXPR << std::endl; \
} while (0)
 ```

Then these bug-prone output copypastas can become simply
```c
NOISY_WARNING( "string [" << token << "] did not convert to int as expected" );
```

Comment (ejwilson3) [inline MCNPInput.cpp]: It was actually decided to remove the cerr outputs entirely; those changes will be in the next PR, but I could make another small commit to this one removing them as well.

Comment (sjackso) [inline MCNPInput.cpp]: Don't bother -- if you're planning to clean it up in in a future PR that's fine.

(Output and logging in c++ is always a colorful bikeshed.  In the past I've written custom ostream subclasses that can optionally direct to zero, one, or multiple outputs (stdout, file, etc.), but it's a hassle to maintain.  Lavtely I've found it most comfortable to create macros that accept ostream syntax.  This provides a nice way to centrally control verbosity, write to multiple outputs, compile out DEBUG statements, etc.)

File: MCNPInput.hpp [hunk 2]
```diff
@@ -3,12 +3,14 @@
 
 #include <vector>
 #include <map>
-#include <iosfwd>
+//#include <iosfwd>
```
Comment (pshriwise) [inline MCNPInput.hpp]: Can these be removed entirely rather than being commented?

**Comment by gonuke** [comment]
Can we move the CLI files into a subdirectory - perhaps called `cli` with appropriate changes to the `CMakeLists.txt` file?

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+if(CMAKE_SYSTEM_NAME MATCHES Linux AND CMAKE_COMPILER_IS_GNUCXX)
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "5.0")
+    if(NOT CMAKE_CXX_FLAGS MATCHES _GLIBCXX_USE_CXX11_ABI)
+      set(CMAKE_CXX_FLAGS "-D_GLIBCXX_USE_CXX11_ABI=0 ${CMAKE_CXX_FLAGS}")
+    endif()
+  endif()
+endif()
+
+# Get the packages needed to use the SDK
+find_package(Cubit REQUIRED CONFIG)
```
Comment (gonuke) [inline CMakeLists.txt]: Why is this required?  Do we perhaps need a more generic version of this to find some form of `iGeom`?

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+include_directories(${CUBIT_INCLUDE_DIRS})
+
+# These directories need to be defined by the user.
+include_directories(${IGEOM_INCLUDE_DIR})
+include_directories(${MOAB_INCLUDE_DIR})
```
Comment (gonuke) [inline CMakeLists.txt]: Why do we need MOAB includes?  I thought this depend only iGeom?

**Comment by gonuke** [comment]
We probably need to revise the README to reflect these changes.

File: CMakeLists.txt (hunk 3 above)
Comment (ejwilson3) [inline CMakeLists.txt]: There are a number of header files needed by iGeom that are found in Trelis. iGeom.h needs to include RefGroup.hpp for much of the CATag parts, which in turn includes more files.
mcnp2cad also needs GeometryModifyTool.hpp (which include I had erroneously put into iGeom.h) in order to take advantage of the PRINT_INFO macro, without which we would have to find another way to print to the Trelis command line, if we want to do that.

Comment (gonuke) [inline CMakeLists.txt]: but this will now be a stand-alone library that needs to know where to find iGeom and hopefully upon doing so, is told where to find the other includes

File: CMakeLists.txt (hunk 4 above)
Comment (ejwilson3) [inline CMakeLists.txt]: I had been linking to the MOAB includes for iBase.h, since iGeom uses iBase_EntityHandles to keep track of everything.  I can actually just take iBase.h from MOAB; it's only about 500 lines. I was trying to avoid copying files as much as possible, but it's probably worth it to avoid having to have MOAB installed as well if one isn't using the plugin's iGeom.

Comment (gonuke) [inline CMakeLists.txt]: Doesn't iGeom have it's own iBase?

File: CMakeLists.txt (hunk 3 above)
Comment (gonuke) [inline CMakeLists.txt]: per our discussion: change `Cubit` to `CGM` and then use `CGM_*_DIR` throughout.

File: CMakeLists.txt (hunk 4 above)
Comment (gonuke) [inline CMakeLists.txt]: Insert iBase into iGeom like https://bitbucket.org/fathomteam/cgm/src/34e0c6581eff1ef5be0ce17ac815bf1ffde9c844/itaps/iGeom.h?at=master&fileviewer=file-view-default

**Comment by gonuke** [comment]
Replaced by #53
